### PR TITLE
feat(frontend): use the Sifi API for getting token balances

### PR DESCRIPTION
**Pull request by boriskubrik**
- Replace the multicall to get the token balances with using the API provided by Sifi.
- Store all `balanceMaps` in state, using the `chainId` as a key. We fetch once and get the balances for all tokens.
- Stale time is set to 1 minute. The refetching logic remains the same, handled by the `useRefetchBalances` hook. 
- `isLoading` is stored in state. CoinSelector does not yet support it for the balances, this needs to be added to shared-ui.
- Once we have the the loading state, we can move the first fetch to when the user has opened the TokenSelector. For now, it would look bad to show the UI without any balances.
- Displaying `0` for coins without balances. Not displaying anything while we are fetching the balances. This should be replaced with the loading state.

image

### CODE REVIEW

File: packages/frontend/src/providers/WalletBalancesProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+              return { chainId, tokenId, formattedBalance, usdValue };
+            }
+          })();
+        });
+
+        Promise.all(balancePromises)
+          .then(balances => {
+            const newBalanceMapsByChain: Record<number, BalanceMap> = {};
+
+            balances.forEach(balance => {
```
Comment (xykota) [inline packages/frontend/src/providers/WalletBalancesProvider.tsx:87]:
```typescript
const newBalanceMapsByChain = balances.reduce<Record<number, BalanceMap>>((acc, balance), {
  if (!balance) {
    console...
    return acc;
  }

  const map = acc[chainId] ?? new Map();

  map.set(...);

  return {
    ...acc,
    [chainId]: map,
  };
}, {});
```

**Review by xykota — DISMISSED** [review]
Minor

**Review by immutablegoblin — CHANGES_REQUESTED** [review]
Pretty heavy console spamming from failed Coingecko lookups here

image

We should probably handle these better by aggregating or suppressing them, going to make debugging a chore in the future.

**Comment by immutablegoblin** [comment]
I also can't work out whats going on here but something isn't right:

When I switch to Avalance we're trying to run `balanceOf` on this contract, but it's not a token - is this one of ours?

https://etherscan.io/address/0x2791bca1f2de4661ed88a30c99a7a9449aa84174#multichain-portfolio

My understanding is we're only supposed to be running `balanceOf` directly on the currently selected `to` and `from` tokens for the most up to date balance, so this seems unexpected.

**Comment by immutablegoblin** [comment]
Oh I think I worked it out.

Currently if you change networks it will take the previously selected token and then try and fetch the balance for the _old token address on the new network_ resulting in a failure.

**Comment by boriskubrik** [comment]
@immutablegoblin I've suppressed the Coingecko errors.  [e3bd912](https://github.com/sifiorg/sifi/pull/527/commits/e3bd9123486483d03512e81378392b0d1718b3d0) 

We still see errors from the SDK, but it is a separate matter.

**Review by immutablegoblin — APPROVED** [review]

**Pull request merged**